lldb/source/Commands/CommandObjectSource.cpp

```diff
@@ -1067,7 +1067,14 @@ class CommandObjectSourceList : public CommandObjectParsed {
                 &result.GetOutputStream(), m_options.num_lines,
                 m_options.reverse, GetBreakpointLocations())) {
           result.SetStatus(eReturnStatusSuccessFinishResult);
+        } else {
+          if (target.GetSourceManager().AsLastLine(m_options.reverse)) {
```

Comment:
If you go in reverse and try to page past the beginning of the file we really shouldn't print "Reached the end to the file".  Maybe:

result.AppendNoteWithFormat("Reached {0} of the file, no more to page", m_options.reverse ? "beginning" : "end");